basedtyping/__init__.py

```diff
@@ -556,3 +558,44 @@ def f[T](t: TypeForm[T]) -> T: ...
                             
                              reveal_type(f(int | str))  # int | str
                          """)
+
+
+@dataclass
+class _BaseGenericFunction(Generic[P, T]):
+    fn: Callable[P, T]
+
+
+@dataclass
+class _GenericFunction(_BaseGenericFunction[P, T]):
+    # TODO: make this an TypeVarTuple when mypy supports it
+    #  https://github.com/python/mypy/issues/16696
+    parameters: tuple[object, ...] | None = None
```

Comment:
maybe add a comment saying that this isnt used yet and link to the issue for reifying them